WebSharper.UI/Attr.Client.fs

```diff
@@ -417,6 +418,30 @@ module BindVar =
     let FloatApplyChecked : Apply<CheckedInput<float>> =
         ApplyValue FloatGetChecked FloatSetChecked
 
+    let DecimalSetUnchecked : Set<decimal> = fun el i ->
+        el?value <- string i
+    let DecimalGetUnchecked : Get<decimal> = fun el ->
+        let s = el?value
+        if String.isBlank s then Some 0.m else
+        let pd : decimal = JS.Plus s

```

Comment:
Hello @diegopego! In this scenario I think we should use either MathJS to construct the decimal object or use System.Decimal.TryParse to ensure we are ending up with a MathJS representation, as currently the above code would just parse it to a JS Number, which could cause unexpected issues otherwise.

Other than this + what András looked above, the PR looks good to me

Reply:
Is there a reason not to use TryParse for all the types?
	- FloatGetChecked uses JS.Plus
	- IntGetChecked uses TryParse

I've tried System.Decimal.TryParse and it raised this error: 

`Method name not found in JavaScript compilation: (TryParse : System.String * byref<System.Decimal> -> System.Boolean), Members: mathn, un, bin, mul, Abs, Add, Ceiling, Compare, CompareTo, Divide, Equals, GreaterThan, LessThan, Floor, Max, Min, Multiply, Parse, Remainder, Subtract, get_Sign, get_Zero, op_Addition, op_Division, op_Equality, op_Inequality, op_GreaterThan, op_LessThan, op_GreaterThanOrEqual, op_LessThanOrEqual, op_Modulus, op_Multiply, op_Subtraction, op_UnaryNegation, op_UnaryPlus`

```
let DecimalGetUnchecked : Get<decimal> = fun el ->
	let s = el?value
	if String.isBlank s then Some 0.m
	else
		match System.Decimal.TryParse(s) with
			| (true, result) -> Some result
			| (false, _) -> None
```

Reply:
Right it indeed seems like we are missing the TryParse macro, but we have the normal Parse for Decimal. For the time being you could wrap that Parse (it'd just casting a MathJS bignumber instance to decimal) in a try catch and in case of error, return None. I'll add a ticket to make sure we have a TryParse macro for decimal

For Int and Float, we can just rely on JS.Plus, we don't need the TryParse functions in those scenarios.

Reply:
@Jooseppi12 Decimal proxy does not use macro as it's differs from other numeric types by using MathJS, so you can just add proxy method next to 

https://github.com/dotnet-websharper/core/blob/09eec1ed993f8e1a5f23b140782d96f1f894fec4/src/stdlib/WebSharper.MathJS.Extensions/Decimal.fs#L169-L170

Reply:
@Jooseppi12 

I tried the following, but now, I'm having another error on the browser:
```
Uncaught Error: Cannot compare function values.
    at Operators.FailWith (Operators.fs:223:42)
    at Unchecked.Compare (Unchecked.fs:81:13)
    at Unchecked.fs:65:24
    at Unchecked.Compare (Unchecked.fs:65:64)
    at Client.ValidateCheckedDecimalPositive (Client.fs:56:15)
    at Forms.fs:534:31
    at Snap.fs:299:48
    at Snap.When (Snap.fs:201:57)
    at Snap.Map (Snap.fs:299:13)
    at Object.o (Reactive.fs:233:13)
```

```
let DecimalGetUnchecked : Get<decimal> = fun el ->
        //let s = el?value
        //if String.isBlank s then Some 0.m else
        //let pd : decimal = JS.Plus s
        //if JS.IsNaN pd then None else Some pd

        let s = el?value
        if String.isBlank s then Some (MathJS.Math.Bignumber(0.0m) |> As<decimal>) else
        try
            let pd : decimal = MathJS.Math.Bignumber(s) |> As<decimal>
            Some pd
        with
        | _ -> None

let DecimalGetChecked : Get<CheckedInput<decimal>> = fun el ->
            //let s = el?value
            //if String.isBlank s then
            //    if CheckValidity el then Blank s else Invalid s
            //else
            //    let i = JS.Plus s
            //    if JS.IsNaN i then Invalid s else Valid (i, s)
            //|> Some
            let s = el?value
            if String.isBlank s then
                if CheckValidity el then Blank s else Invalid s
            else
                try
                    let i : decimal = MathJS.Math.Bignumber(s) |> As<decimal>
                    Valid (i, s)
                with
                | _ -> Invalid s
            |> Some
```

Reply:
What does your `Client.ValidateCheckedDecimalPositive` do?

Reply:
```
let ValidateCheckedDecimalPlaces places (f:CheckedInput<decimal>) =
        match f with
        | Valid(value, inputText) ->
            let r = Math.Round(value, places)
            r = value
        | Invalid _ -> false
        | Blank _ -> false
```

```
<*> (Form.Yield (CheckedInput.Make 0.0m)
            |> Validation.Is (fun x -> ValidateCheckedDecimalPlaces 2m x) "Price must have up to two decimal places"
            )
```

Reply:
MathJS.Math.Bignumber fixed the equality issue that occurred with this code: ` Math.Round(value, places) = value`

I figured out that `Uncaught Error: Cannot compare function values.` is related to ValidateCheckedDecimalPositive.

On the client, I have these validate functions:
```
let ValidateCheckedDecimalPositive (f:CheckedInput<decimal>)=
    match f with
    | Valid(value, inputText) -> value > 0.0m
    | Invalid _ -> false
    | Blank _ -> false

let ValidateCheckedDecimalPlaces places (f:CheckedInput<decimal>) =
    match f with
    | Valid(value, inputText) ->
        let r = Math.Round(value, places)
        r = value
    | Invalid _ -> false
    | Blank _ -> false
```

With this DecimalGetChecked  version
- `Math.Round(value, 2) = value` don't work as expected
- `value > 0.0m` works
```
let DecimalGetChecked : Get<CheckedInput<decimal>> = fun el ->
    let s = el?value
    if String.isBlank s then
        if CheckValidity el then Blank s else Invalid s
    else
        let i = JS.Plus s
        if JS.IsNaN i then Invalid s else Valid (i, s)
    |> Some
```

When using DecimalGetChecked  with Bignumber:
- `Math.Round(value, 2) = value`  works as expected
- `value > 0.0m` gives `Uncaught Error: Cannot compare function values.`
- `let isGreater = MathJS.Math.Larger(value, 0.0m) |> As<bool>` works as expected

```
let DecimalGetChecked : Get<CheckedInput<decimal>> = fun el ->
    let s = el?value
    if String.isBlank s then
        if CheckValidity el then Blank s else Invalid s
    else
        try
            let i : decimal = MathJS.Math.Bignumber(s) |> As<decimal>
            Valid (i, s)
        with
        | _ -> Invalid s
    |> Some
```

Reply:
Yeah, the catch is we need the MathJS.Bignumber based version in WebSharper.UI, otherwise we have no guarantee that the decimal representation is correct.

The issue in the Validate function comes from the core repo, as we have a Macro that handles special translation around the `<` and `>` operators, ignoring their proxies defined.

So feel free to commit the MathJS based DecimalGetChecked function to this PR and I will take care of this issue in the core repo (+ add the above mentioned missing TryParse proxy)